tests/framework/artifacts.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Define classes for interacting with CI artifacts in s3."""
+
+import os
+import platform
+import re
+
+from shutil import copyfile
+from typing import List
+from os import path
+from pathlib import Path
+
+import boto3
+import botocore.client
+
+
+MICROVM_CONFIG_EXTENSION = ".json"
+MICROVM_KERNEL_EXTENSION = ".bin"
+MICROVM_DISK_EXTENSION = ".ext4"
+
+
+class Artifact:
+    """A generic artifact manipulation class."""
+
+    bucket = None
+    key = None
+    local_folder = None
+
+    def __init__(self, bucket, key, type="misc"):
+        """Initialize bucket, key and type."""
+        self.bucket = bucket
+        self.key = key
+        self.type = type
+
+    def name(self):
+        """Return the artifact name."""
+        return self.key.rsplit("/", 1)[1]
+
+    def local_dir(self):
+        """Return the directory containing the downloaded artifact."""
+        return "{}/{}/{}".format(
+            self.local_folder,
+            self.type,
+            platform.machine(),
+        )
+
+    def download(self, target_folder, force=False):
+        """Save the artifact in the folder specified target_path."""
+        self.local_folder = target_folder
+        Path(self.local_dir()).mkdir(parents=True, exist_ok=True)
+        if force or not os.path.exists(self.local_path()):
+            self.bucket.download_file(self.key, self.local_path())
+
+    def local_path(self):
+        """Return the local path where the file was downloaded."""
+        # The file path format: <target_folder>/<type>/<platform>/<name>
+        return "{}/{}".format(
+            self.local_dir(),
+            self.name()
+        )
+
+
+class DiskArtifact(Artifact):
+    """Specializes the generic artifact."""
+
+    def ssh_key(self):
+        """Return a ssh key artifact."""
+        key_file_path = self.key.rsplit(".", 1)[0] + '.id_rsa'
```

Comment:
Same proposal here, as above. You could use `Path` to make the purpose of statement clearer.

Reply:
done